Correctionsvisualisations

jujusb opened the pull request.

Commit:
Modification of bugs of not detecting videos in xAPI or good Answer (addition of the count) if the success is the boolean True and not only the text "True”

Commit:
Adapt each graph to each max value and not a fixed value +  addition of a condition if it's nothing to show

Commit:
corrections of a actuals visualisation : if object_type_xapi in ["alternative", "Alternative"] to have in count maj + xAPISG-CorrectIncorrectPlayer.ipynb limit actualized + xAPISG-AlternativesSelectedQuestion.ipynb variable i not initialized

Commit:
Correction of the transformation of the timestamp in datetime in the processStatement File and not in the visualisation file

gorco commented on vis/xAPISG-CompletablesProgress.ipynb:
```diff
@@ -25,6 +25,8 @@
     "    # progress of each player per completable\n",
     "    players_progress = []\n",
+    "    scoremax=-float('Inf')\n",
+    "    scoremin=float('Inf')\n",
```
Please, can you change the var names? The visualization is about progress and not about score

```suggestion
    "    progressmin=float('Inf')\n",
```

jujusb replied:
Corrected in the last commit

gorco commented on vis/xAPISG-CompletablesProgress.ipynb:
```diff
@@ -42,8 +42,12 @@
     "            players_progress.append(player_progress)\n",
     "            \n",
+    "    if scoremin!=0:\n",
+    "        scoremin-=1\n",
```
This is the minimun progress of the user and it can never be _<0_. 
E.g: If "scoremin" is _0.1_ "scororemin" will be _-1.1_

Also, please, change the var. name, the visualization is about the user progress. This name confuses

gorco commented on vis/xAPISG-CompletablesProgress.ipynb:
```diff
@@ -45,19 +50,21 @@
     "    index = np.arange(n_groups)\n",
     "    bar_width = 0.2\n",
     "    opacity = 0.8\n",
```
This variable is never used

jujusb replied:
Corrected in the last commit

gorco commented on vis/xAPISG-CompletablesProgress.ipynb:
```diff
@@ -33,31 +35,40 @@
+    "        \n",
+    "        plt.legend()\n",
+    "        plt.xlabel(\"Completable\")\n",
+    "        plt.ylabel(\"Progress\")\n",
```
I think we should specify the range of values it can take, p.g. "Progress (0 to 1)" and also, I think that is better if we use % in all visualizations like this, i.e use 0% to 100%. If the value is 0.75 we have to transform to 75. 
What do you think @crisal24 ?

gorco commented on ProcessxAPISGStatement.ipynb:
```diff
@@ -113,18 +123,19 @@
-    "                    player_info[\"alternatives\"][object_id_name].append((result_response, result_success))\n",
-    "                else:\n",
-    "                    player_info[\"alternatives\"][object_id_name] = [(result_response, result_success)]\n",
+    "            if object_type_xapi in [\"Alternative\",\"alternative\"]:\n",
```
I think is better use .lower()
`if object_type_xapi.lower() =="alternative"`
And same in other cases. The change allows to work with the word "alternative" regardless of how it is written (upper,  lower or capital case)

gorco commented on ProcessxAPISGStatement.ipynb:
```diff
@@ -91,13 +91,23 @@
     "                player_info[\"game_started\"] = True\n",
     "            if timestamp:\n",
-    "                player_info[\"completables_times\"][object_id_name] = timestamp\n",
+    "                for timeformat in ('%Y-%m-%dT%H:%M:%SZ', '%Y-%m-%dT%H:%M:%S.%fZ'):\n",
```
I understand that you want to check diferents time formats. I see the time format like a configurable variable (maybe an array of formats) that user can change easely like the traces file directory. And also we can transform this block of code in a function that return the value if any format match and show a warning/exception if not

jujusb replied:
Corrected in the last commit

gorco commented on vis/xAPISG-CompletablesScores.ipynb:
```diff
@@ -41,18 +41,19 @@
     "    index = np.arange(n_groups)\n",
     "    bar_width = 0.2\n",
     "    opacity = 0.8\n",
```
Not used

jujusb replied:
Corrected in the last commit

gorco commented on vis/xAPISG-PlayersProgress.ipynb:
```diff
@@ -16,21 +16,24 @@
-    "    plt.gca().xaxis.set_major_locator(mdates.MinuteLocator(interval=5))\n",
-    "    plt.title(\"Progress of players by time\")\n",
-    "    plt.show()"
+    "            #print(progress_time)\n",
```
Remove the commented code, please

jujusb replied:
Corrected in the last commit

gorco reviewed (changes requested):
Review completed. Good job. 

I've left some comments in the code and other following to improve the code and visualizations:
I think we can improve some of the visualizations by adding options. In the case of the "CorrectIncorrectPlayer", "CorrectIncorrectQuestion" and "VideosSeenSkiped" displays we could add the option of a "normalized", i.e. being the Y-axis the percentage from 0 to 100. That can be useful to know which question has a higher error rate regardless of whether it appears more or less.

Anyway, we have several problems with the displays that we should solve before create more visualizations. Although there is no need to make changes in this PR, it is necessary to address it as soon as possible. 
- If there is no data to fill a visualization nothing is shown. In this case at least some message should be shown.
- There are cases in which if there is a lot of data, options or users, the display is unable to show them and also the bars are not adjusted. I think it is best to meet to discuss the issue and decide how to address this problem.

Commit:
Corrections score to progress in completableProgress + remove comment and variables not used + add definition of functions + move the legend to the side of the graph + addition of a message if it is nothing to show + function to transform the timestamp in datetime + addition of an array of timeformat in the Processor

Commit:
show the display box to show inside the message "no data to fill the visualisation" + addition of a label to some axes + modification to the progress to have it in % and only taking in count values in 0 to 1.

Commit:
Change the in [] to better use .lower() => for exemple object_type.xapi.lower()=="alternative" and same in other cases.
This change allows to work with the word "alternative" regardless of how it is written (upper, lower or capital case).

Commit:
Revert "Change the in [] to better use .lower() => for exemple object_type.xapi.lower()=="alternative" and same in other cases."

This reverts commit c81daee21a1a50920b4d358e771060c102a7d9e1.

Commit:
Function if another file to starting show the display box with inside the message "no data to fill the visualisation" + correction function ProgressCompletable

Commit:
Change the in [] to better use .lower() => for exemple object_type.xapi.lower()==alte
